### PR TITLE
chore: lint JSDoc with `eslint-plugin-jsdoc`

### PR DESCRIPTION
**Pull request by danvk**
This is helpful for enforcing some norms around JSDoc comments. See https://github.com/refstudio/refstudio/pull/105/files#discussion_r1223202950

In particular:

```ts
/**
 * short comment
 */
```

gets auto-fixed to the one-line form:

```ts
/** short comment */
```

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-undef */
```
Comment (danvk) [inline .eslintrc.cjs:1]: I got a warning about this file not being included in `tsconfig.json`, so I fixed that. That resulted in prettier and eslint running over this, which standardized the quotes and formatting.

File: .eslintrc.cjs [hunk 2]
```diff
@@ -128,12 +135,45 @@ module.exports = {
...
+    ],
+    'jsdoc/multiline-blocks': [
+      'error',
+      {
+        noZeroLineText: true,
+        noMultilineBlocks: true,
+        minimumLengthForMultiline: 80,
+      },
+    ],
+    'jsdoc/no-restricted-syntax': [
```
Comment (danvk) [inline .eslintrc.cjs:161]: This disallows JSDoc comments in various contexts where they don't make sense.

File: src/atoms/referencesState.ts [hunk 3]
```diff
@@ -5,15 +5,11 @@ import { ReferenceItem } from '../types/ReferenceItem';
 
 type ReferencesState = Record<string, ReferenceItem>;
 
-/**
- * INTERNAL ATOMS
- */
+/** INTERNAL ATOMS */
```
Comment (danvk) [inline src/atoms/referencesState.ts:8]: These got shortened to the one-line form. But really they shouldn't be JSDoc comments at all (they could just be inline comments, or `/* ... */` comments -- not jsdoc).

File: src/hooks/useDebouncedCallback.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@ import { useCallback, useRef } from 'react';
 
 /**
  * Returns a callback that will only call the passed function when it has not been called for the specified time
+ *
```
Comment (danvk) [inline src/hooks/useDebouncedCallback.ts:5]: This is `jsdoc/tag-lines`. I'd like to enforce something like [PEP 257](https://peps.python.org/pep-0257/#multi-line-docstrings): a summary line followed by a blank line, followed by a longer description. eslint-plugin-jsdoc used to have a rule that did that but they recently deleted it https://github.com/gajus/eslint-plugin-jsdoc/issues/1031#issuecomment-1583129432

**Review by sehyod — DISMISSED** [review]

**Review by cguedes — DISMISSED** [review]

**Pull request merged**